lib/plugins/Label_H1_FPN.ts

```diff
@@ -0,0 +1,158 @@
+import { decode } from 'base85';
+import { DateTimeUtils } from '../DateTimeUtils';
+import { DecoderPlugin } from '../DecoderPlugin';
+import { CoordinateUtils } from '../utils/coordinate_utils';
+import { RouteUtils } from '../utils/route_utils';
+
+export class Label_H1_FPN extends DecoderPlugin {
+  name = 'label-h1-fpn';
+
+  qualifiers() { // eslint-disable-line class-methods-use-this
+    return {
+      labels: ["H1"],
+      preambles: ['FPN'],
+    };
+  }
+
+  decode(message: any, options: any = {} ): any {
+    let decodeResult: any = this.defaultResult;
+    decodeResult.decoder.name = this.name;
+    decodeResult.formatted.description = 'Flight Plan';
+    decodeResult.message = message;
+
+    const checksum = message.text.slice(-4);
+    const data = message.text.slice(0, message.text.length-4).split(':');
+
+    if(data[0].startsWith('FPN/')) {
+      let allKnownFields = parseHeader(decodeResult, data[0]);
+      for(let i=1; i< data.length; i+=2) {
+        const key = data[i];
+        const value = data[i+1];
+        // TODO: discuss how store commented out bits as both raw and formatted
```

Comment:
i should probably discuss before putting out a PR, but sometimes i find it easier when there's code to point at